### ISSUE TITLE
"Next Bill On" bug in MyAccount for Memberships/Subscriptions Integration

### ISSUE THREAD

**Issue by youllbejustfine** [woocommerce/woocommerce#14579]
### EXPLANATION OF THE ISSUE

The "Next Bill On" column is not outputting in the My Account dashboard. Issue has been traced to the function **my_memberships_subscriptions_columns()** on **line 111** of **class-wc-memberships-integration-subscriptions-frontend.php**.

The array_splice function does not create the key "membership-next-bill-on" as expected, but instead will place the new item into the 'wc_memberships_my_memberships_column_names' $columns array with a key of [0] (please see http://php.net/manual/en/function.array-splice.php#56794 for documentation of this behavior). Because of this, the **add_action('wc_memberships_my_memberships_column_membership-next-bill-on')** on **line 49** never gets called.

A better approach would be to use array_merge in order to successfully create the array key. See the proposed code below: 

```
public function my_memberships_subscriptions_columns( $columns ) {

	// Insert before the 'Actions' column.		
	$columns = array_merge(array_splice($columns, 0, 4), array(
		'membership-next-bill-on' => __( 'Next Bill On', 'woocommerce-memberships' ),
	), $columns);

	return $columns;
}
```
And as a result, the wrapping **td** tags can be removed in function **output_subscription_columns()**

### STEPS TO REPRODUCE THE ISSUE

Running Environment:
Wordpress v4.7.3
WooCommerce v3.0.3
WooCommerce Memberships v1.8.2
WooCommerce Subscriptions v2.2.4

Create a WooCommerce Member with an Active Subscription, and view the MyAccount dashboard page.

### SYSTEM STATUS

<details>
```
`
### WordPress Environment ###

Home URL: http://inroads.lukassteinman.com
Site URL: http://inroads.lukassteinman.com
WC Version: 3.0.3
Log Directory Writable: ✔
WP Version: 4.7.3
WP Multisite: –
WP Memory Limit: 1 GB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: ❌ 5.4.45 - We recommend a minimum PHP version of 5.6. See: How to update your PHP version
PHP Post Max Size: 64 MB
PHP Time Limit: 300
PHP Max Input Vars: 10000
cURL Version: 7.45.0
OpenSSL/1.0.1e

SUHOSIN Installed: –
MySQL Version: 5.6.33
Max Upload Size: 64 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.0.3
WC Database Prefix: wp_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ✔

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (21) ###

ACF QuickEdit Fields: by Jörn Lund – 1.1.1
Add From Server: by Dion Hulse – 3.3.3
Admin Menu Editor: by Janis Elsts – 1.7.3
Advanced Custom Fields PRO: by Elliot Condon – 5.5.11
Cimy User Manager: by Marco Cimmino – 1.5.0
Inroads Featured Articles: by Lukas Steinman – 1.0.0
User Profile Picture: by Ronald Huereca – 1.4.3
Post Type Switcher: by John James Jacoby – 3.0.0
Post Views Counter: by dFactory – 1.2.7
Regenerate Thumbnails: by Alex Mills (Viper007Bond) – 2.2.6
Social Pug: by DevPups – 2.0.3
SVG Support: by Benbodhi – 2.3.7
Theme My Login: by Jeff Farthing – 6.4.9
Velvet Blues Update URLs: by VelvetBlues.com – 3.2.7
WooCommerce Memberships: by SkyVerge – 1.8.2
WooCommerce Subscriptions Gifting: by Prospress Inc. – 1.0
WooCommerce Subscriptions: by Prospress Inc. – 2.2.4
WooCommerce: by Automattic – 3.0.3
WooCommerce Helper: by WooCommerce – 1.7.2 – Network enabled
WordPress Importer: by wordpressdotorg – 0.6.3
WP LESS: by Oncle Tom – 1.8.0

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: CAD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 0
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
subscription (subscription)
variable (variable)
variable subscription (variable-subscription)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #9449 - /subscribe/
Cart: #9450 - /cart/
Checkout: #9451 - /checkout/
My account: #9452 - /my-account/

### Theme ###

Name: Inroads
Version: 1.0.0
Author URL: http://lukassteinman.com
Child Theme: ❌ – If you're modifying WooCommerce on a parent theme you didn't build personally
then we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: inroads/woocommerce/archive-product.php
inroads/woocommerce/content-product.php
inroads/woocommerce/loop/add-to-cart.php
inroads/woocommerce/loop/price.php
inroads/woocommerce/myaccount/my-account.php
inroads/woocommerce/single-product.php


### subscriptions ###

WCS_DEBUG: ✔ No
Subscriptions Mode: ✔ Live
`
```
</details>

**Comment by mikejolley** [comment]
@bekarice

**Comment by bekarice** [comment]
@youllbejustfine please submit issues for plugins via [the WooCommerce help desk](http://woocommerce.com/my-account/tickets/), as this is a core bug tracker.

This is fixed in the upcoming v1.8.4 release, and opening a thread via the help desk provides the ability to get a pre-release version as well. thanks!
